semantic_release/vcs_helpers.py

```diff
@@ -37,18 +37,34 @@ def get_formatted_tag(version):
     return tag_format.format(version=version)
 
 
-def get_commit_log(from_rev=None):
+def get_commit_log(from_rev=None, to_rev=None):
     """Yield all commit messages from last to first."""
-    rev = None
     if from_rev:
         from_rev = get_formatted_tag(from_rev)
         try:
             repo().commit(from_rev)
-            rev = f"...{from_rev}"
         except BadName:
+            from_rev = None
             logger.debug(
                 f"Reference {from_rev} does not exist, considering entire history"
             )
```

Comment:
Generate the log message before changing the variable
```suggestion
            logger.debug(
                f"Reference {from_rev} does not exist, considering entire history"
            )
            from_rev = None
```

Reply:
done